src/core/varlink-unit.c

```diff
@@ -414,7 +416,22 @@ int vl_method_list_units(sd_varlink *link, sd_json_variant *parameters, sd_varli
                 return list_unit_one(link, unit, /* more = */ false);
         }
 
-        // TODO lookup by invocationID
+        if (!sd_id128_is_null(p.invocation_id)) {
+                Unit *unit = hashmap_get(manager->units_by_invocation_id, &p.invocation_id);
+                if (!unit)
+                        return sd_varlink_error(link, VARLINK_ERROR_UNIT_NO_SUCH_UNIT, NULL);
+
+                /* XXX DBus does the following here:
+                 * mac_selinux_unit_access_check(u, message, "status", error)
+                 *
+                 * Btw, same for GetUnitByPIDFD (which I missed in previous PR),
+                 * but it doesn't do it for GitUnitByPID.
```

Comment:
probably should do the check for all cases the same.

Reply:
so, I just mimic dbus even though it doesn't make sense to me (or I don't understand it)?

Reply:
ignore my previous message. I misread your statement

Reply:
I've looked into implementing SELinux checks for varlink. It's not straightforward but I can do it. The main challenge for me is how to test it. Any hints/guidance?